src/_P099_XPT2046Touch.ino

```diff
@@ -0,0 +1,450 @@
+
+  switch (function)
+  {
+
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_099;
+        Device[deviceCount].Type = DEVICE_TYPE_SPI;
+        Device[deviceCount].VType = SENSOR_TYPE_TRIPLE;
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = false;
+        Device[deviceCount].ValueCount = 3;
+        Device[deviceCount].SendDataOption = false;
+        Device[deviceCount].TimerOption = false;
+        success = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:
+      {
+        string = F(PLUGIN_NAME_099);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+      {
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_099));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_099));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_099));
+        success = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEGPIONAMES:
+      {
+        event->String1 = formatGpioName_output(F("TS CS"));
+        break;
+      }
+
+    case PLUGIN_SET_DEFAULTS:
+      {
+        byte init = P099_CONFIG_STATE;
+
+        // if already configured take it from settings, else use default values
+        if(init != 1)
+        {
+          P099_CONFIG_CS_PIN      = P099_TS_CS;
+          P099_CONFIG_TRESHOLD    = P099_TS_TRESHOLD;
+          P099_CONFIG_ROTATION    = P099_TS_ROTATION;
+          P099_CONFIG_X_RES       = P099_TS_X_RES;
+          P099_CONFIG_Y_RES       = P099_TS_Y_RES;
+          P099_CONFIG_OBJECTCOUNT = P099_INIT_OBJECTCOUNT;
+
+          uint32_t lSettings = 0;
+          bitWrite(lSettings, P099_FLAGS_SEND_XY,         P099_TS_SEND_XY);
+          bitWrite(lSettings, P099_FLAGS_SEND_Z,          P099_TS_SEND_Z);
+          bitWrite(lSettings, P099_FLAGS_SEND_OBJECTNAME, P099_TS_SEND_OBJECTNAME);
+          bitWrite(lSettings, P099_FLAGS_USE_CALIBRATION, P099_TS_USE_CALIBRATION);
+          bitWrite(lSettings, P099_FLAGS_LOG_CALIBRATION, P099_TS_LOG_CALIBRATION);
+          P099_CONFIG_FLAGS  = lSettings;
+        }
+        success = true;
+        break;
+      }
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        addFormSubHeader(F("Screen"));
+
+        uint16_t width_ = P099_CONFIG_X_RES;
+        if(width_ == 0)
+          width_ = P099_TS_X_RES; // default value
+        addFormNumericBox(F("Screen Width (px) (x)"), F("p099_width"), width_, 1, 65535);
+
+        uint16_t height_ = P099_CONFIG_Y_RES;
+        if(height_ == 0)
+          height_ = P099_TS_Y_RES; // default value
+        addFormNumericBox(F("Screen Height (px) (y)"), F("p099_height"), height_, 1, 65535);
+
+        byte choice2 = P099_CONFIG_ROTATION;
+        String options2[4] = { F("Normal"), F("+90°"), F("+180°"), F("+270°") };

```

Comment:
Please don't use unicode in the source code as encoded character in the string.
If we need to have a degree symbol, then we should add a function for it.
That may also provide the right encoding either for html (default) or if needed at all one encoded to be sent via serial or something.

Reply:
I 'borrowed' that from P095...

Reply:
Then that one also needs some de-Unicoding :)